### PR TITLE
Add script to check build reproducibility

### PR DESCRIPTION
**Pull request by tiziano88**
Ref #241

### CODE REVIEW

File: scripts/check_enclave_reproducibility [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+# This script builds the Asylo server twice from scratch (cleaning the Bazel cache before each
+# compilation), saves the hash of the trusted enclave code to a temporary file, and compares the
+# hashes of the two artifacts.
+# It prints nothing if the hashes match, or a diff in case they don't match.
+
+bazel clean
+rm -rf ./bazel-cache/*
+./scripts/build_server
+./scripts/print_enclave_hash > /tmp/hash_0
```
Comment (daviddrysdale) [inline scripts/check_enclave_reproducibility]: `$TMPDIR` rather than hardcoded `/tmp` ?

File: scripts/check_enclave_reproducibility [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+set -o errexit
+set -o nounset
+set -o xtrace
+
+# This script builds the Asylo server twice from scratch (cleaning the Bazel cache before each
+# compilation), saves the hash of the trusted enclave code to a temporary file, and compares the
+# hashes of the two artifacts.
+# It prints nothing if the hashes match, or a diff in case they don't match.
+
```
Comment (daviddrysdale) [inline scripts/check_enclave_reproducibility]: Maybe leave a TODO as a reminder that there are [lots of things to try varying](http://manpages.ubuntu.com/manpages/cosmic/man1/reprotest.1.html#variations) when testing reproducibility

File: scripts/print_enclave_hash [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o xtrace
+
+# Prints the SHA1 hash of the trusted enclave code.
```
Comment (daviddrysdale) [inline scripts/print_enclave_hash]: Are we going to need this script on its own?  If not, could just make if a function in `check_enclave_reproducibility`.

File: scripts/check_enclave_reproducibility (hunk 1 above)
Comment (tiziano88) [reply scripts/check_enclave_reproducibility]: `$TMPDIR` seems not to be set in our Docker image, not sure why.

File: scripts/check_enclave_reproducibility (hunk 2 above)
Comment (tiziano88) [reply scripts/check_enclave_reproducibility]: Interesting, thanks!

File: scripts/print_enclave_hash (hunk 3 above)
Comment (tiziano88) [reply scripts/print_enclave_hash]: I think it is useful to compare hashes across machines / users, as we have just done on Slack :)

**Pull request merged**